packages/dnb-eufemia/src/extensions/payment-card/components/CardNumber.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import classnames from 'classnames'
+import { createSkeletonClass } from '../../../components/skeleton/SkeletonHelper'
+import P from '../../../elements/P'
+
+interface CardNumberTextProps {
+  cardNumber: string
+  skeleton: boolean
+}
+
+/**
+ * A functional component that displays the text representation of a card number
+ * with optional skeleton styling for loading states.
+ *
+ * @param {CardNumberTextProps} props - The properties for the CardNumberText component.
+ * @param {string} props.cardNumber - The card number to display.
+ * @param {boolean} props.skeleton - Determines if skeleton styling is applied for loading states.
+ * @returns A span element containing the card number with optional skeleton styling.
+ */
+const CardNumberText = ({ cardNumber, skeleton }: CardNumberTextProps) => {
+  return (
+    <span
+      className={classnames(
+        'dnb-payment-card__card__element--wrapper',
+        createSkeletonClass('font', skeleton)
+      )}
+    >
+      <P className="dnb-payment-card__card__numbers">{cardNumber}</P>
+    </span>
+  )
+}
+
+const formatCardNumberRegex = /(?=(?:....)*$)/g
+
+/**
+ * Formats a credit card number by preserving a specified number of trailing digits and
+ * masking the rest with spaces. If no specific number of digits is provided, the card
+ * number is formatted by replacing all spaces or non-numeric characters with a single
+ * space.
+ *
+ * @param {string} cardNumber - The credit card number to format.
+ * @param {number} [digits=8] - The number of trailing digits to preserve in the card number.
+ *                              Defaults to 8 if not specified.
+ * @returns {string} The formatted credit card number.
+ */
+export const formatCardNumber = (
+  cardNumber: string,
+  digits: number
```

Comment:
```suggestion
  digits: number = 8
```

If we go with this, we could also revert these tests, just to make sure that the default behavior now is the same as previous(before this PR), so that we don't break any existing functionalities for apps 🙏 

https://github.com/dnbexperience/eufemia/pull/5085/files#r2095500066
https://github.com/dnbexperience/eufemia/pull/5085/files#r2095499865
https://github.com/dnbexperience/eufemia/pull/5085/files#r2095499628

Reply:
I think this makes more sense, I will apply it 👍

Reply:
I remember removing this, ESLint was triggering an error and suggested removing the annotation since it can get the type from the default value, so it would look like: `digits = 8`

Reply:
Yeah, I guess `digits = 8` is just as good 👍